### PR TITLE
Several code/performance improvements

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

File: handlers/objects.go [hunk 1]
```diff
@@ -964,9 +964,6 @@ func (a *RestAPI) UploadContainerObject(ctx echo.Context, containerID apiserver.
 	}
 
 	defer func() {
```
Comment (roman-khimov) [inline handlers/objects.go:947]: You can move it into the `for` loop above.

**Review by roman-khimov — APPROVED** [review]

File: handlers/api.go [hunk 2]
```diff
@@ -56,14 +57,17 @@ const (
 
 // NewAPI creates a new RestAPI using specified logger, connection pool and other parameters.
 func NewAPI(prm *PrmAPI) *RestAPI {
+	if prm.MaxPayloadBufferSize <= 0 {
+		panic("non-positive payload buffer size limit")
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply handlers/api.go]: done

File: handlers/objects.go (hunk 1 above)
Comment (cthulhu-rider) [reply handlers/objects.go]: done

**Review by tatiana-nspcc — APPROVED** [review]

File: cmd/neofs-rest-gw/config.go [hunk 3]
```diff
@@ -585,11 +585,16 @@ func newNeofsAPI(ctx context.Context, logger *zap.Logger, v *viper.Viper) (*hand
 	}
 
 	apiPrm.ServiceShutdownTimeout = defaultShutdownTimeout
-	apiPrm.MaxObjectSize = int64(ni.MaxObjectSize())
+	if apiPrm.MaxPayloadBufferSize = ni.MaxObjectSize(); apiPrm.MaxPayloadBufferSize == 0 {
```
Comment (smallhive) [inline cmd/neofs-rest-gw/config.go:588]: Is it possible to have zero value max object size in the network?

File: cmd/neofs-rest-gw/config.go [hunk 4]
```diff
@@ -585,11 +585,16 @@ func newNeofsAPI(ctx context.Context, logger *zap.Logger, v *viper.Viper) (*hand
...
+		// default to some heuristic value that in practice should not be needed at all
+		apiPrm.MaxPayloadBufferSize = 4 << 20
```
Comment (smallhive) [inline cmd/neofs-rest-gw/config.go:590]: Could you please move it to the `const` section?

File: handlers/objects.go [hunk 5]
```diff
@@ -956,6 +935,15 @@ func (a *RestAPI) UploadContainerObject(ctx echo.Context, containerID apiserver.
 			resp := a.logAndGetErrorResponse(fmt.Sprintf("get file %q from HTTP request", fileKey), err)
 			return ctx.JSON(http.StatusBadRequest, resp)
 		}
+		defer func() {
+			err := file.Close()
```
Comment (smallhive) [inline handlers/objects.go:939]: Isn't it dangerous to remove `if file == nil` condition?

Comment (smallhive) [inline handlers/objects.go:939]: Isn't it safe to remove the nil check condition?

File: cmd/neofs-rest-gw/config.go (hunk 3 above)
Comment (cthulhu-rider) [reply cmd/neofs-rest-gw/config.go:588]: not really expected, at the same time, there is no statement to the contrary i know about. Anyway, this wouldn't be an app fail point

File: handlers/objects.go (hunk 5 above)
Comment (cthulhu-rider) [reply handlers/objects.go:939]: yeah, safe: we check error above. Or u mean isnt it _unsafe_?

File: cmd/neofs-rest-gw/config.go (hunk 4 above)
Comment (cthulhu-rider) [reply cmd/neofs-rest-gw/config.go]: done

File: cmd/neofs-rest-gw/config.go (hunk 3 above)
Comment (smallhive) [reply cmd/neofs-rest-gw/config.go:591]: I tend to think in the case it is zero, the NeoFS is going to be broken

File: handlers/objects.go (hunk 5 above)
Comment (smallhive) [reply handlers/objects.go:939]: no-no, I mean the `file` is a pointer. Ok

**Review by smallhive — APPROVED** [review]

File: handlers/api.go [hunk 6]
```diff
@@ -56,14 +57,17 @@ const (
 
 // NewAPI creates a new RestAPI using specified logger, connection pool and other parameters.
 func NewAPI(prm *PrmAPI) *RestAPI {
+	if prm.MaxPayloadBufferSize <= 0 {
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply handlers/api.go]: I'm surprised no linter is complaining about it. But `< 0` or `<= 0` comparisons for `uint` can't exist.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**